File: project-euler.java/src/main/java/euler/solutions/_22.java
package euler.solutions;

import euler.common.BaseSolution;

import java.util.Arrays;

public class _22 extends BaseSolution {
    private String[] names;

    @Override
    public void initialize() throws Exception {
        names = dataFileLines();
        Arrays.sort(names);
    }

    @Override
    public long solve() {
        long total_sum = 0;
        int pos = 1;

        for (String name : names) {
            int charSum = 0;
            for (int i = 0; i < name.length(); i++) {
                charSum += name.charAt(i) - 64;
            }
            total_sum += charSum * pos++;
        }

        return total_sum;
    }
}

File: project-euler.go/solutions/_22/22.go
package _22

import (
	"project-euler/common/data"
	"sort"
)

var names []string

func Initialize() {
	names = data.ReadLines()
	sort.Strings(names)
}

func Solve() int64 {
	var total_sum int64 = 0
	pos := 1

	for _,name := range names {
		charSum := 0
		for i := range name {
			charSum += int(name[i]) - 64
		}
		total_sum += int64(charSum * pos)
		pos++
	}

	return total_sum
}
